### PR TITLE
fix: Rename psi, pi and tei  [ TECH-1596 ]

### PR DESCRIPTION
**Pull request by lucaCambi77**
Rename occurrences of _psi_, _pi_, and _tei_ in Java code, in use for SQL alias and variables.

- `psi` renamed to `ev`
- `pi` renamed to `en`
- `tei` renamed to `te`
- `teis` renamed to tracked entities in docs and `trackedEntites` in Java code. In general, renamed with the full name where relevant, for example, in some Java docs.
- clean up in export/event/EventQuery of unused code

**DDL**
- `potentialduplicate`: `teia` and `teib` renamed to `original` and `duplicate`

Potential duplicate endpoint is currently not supported so we align naming

**Not Updated/Out of the scope:**

Some columns could be changed:
- `userteisearchorgunits` (`teiSearchOrganisationUnits` in `User` class) 
- `program` and `trackedentitytype` : `maxteicounttoreturn`
- `RelationshipEntity`, `Authorities` classes stills contains the old enums values

Those changes relate to metadata so it looks complicated to update. We should even discuss if we want to do it

**Not Applied**
- `dhis-services/dhis-service-dxf2` (deprecated)
- `dhis-services/dhis-service-analytics` (https://dhis2.atlassian.net/browse/TECH-1598)
- `dhis-test-e2e` module

### CODE REVIEW

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/trackedentity/TrackedEntityProgramOwnerService.java [hunk 1]
```diff
@@ -40,79 +40,79 @@ public interface TrackedEntityProgramOwnerService {
    * Assign an orgUnit as the owner for a tracked entity instance for the given program. If another
    * owner already exist then this method would fail.
    *
-   * @param teiUid The Uid of the tracked entity instance
+   * @param teUid The Uid of the tracked entity instance
```
Comment (enricocolasante) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/trackedentity/TrackedEntityProgramOwnerService.java:43]:
```suggestion
   * @param teUid The Uid of the tracked entity
```

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/trackedentity/TrackedEntityProgramOwnerService.java [hunk 2]
```diff
@@ -40,79 +40,79 @@ public interface TrackedEntityProgramOwnerService {
...
    */
-  void createTrackedEntityProgramOwner(String teiUid, String programUid, String orgUnitUid);
+  void createTrackedEntityProgramOwner(String teUid, String programUid, String orgUnitUid);
 
   /**
    * Update the owner ou for a tracked entity instance for the given program. If no owner previously
    * exist, then this method will fail.
    *
-   * @param teiUid The tracked entity instance Uid
+   * @param teUid The tracked entity instance Uid
```
Comment (enricocolasante) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/trackedentity/TrackedEntityProgramOwnerService.java:53]:
```suggestion
   * @param teUid The tracked entity Uid
```

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/trackedentity/TrackedEntityProgramOwnerService.java [hunk 3]
```diff
@@ -40,79 +40,79 @@ public interface TrackedEntityProgramOwnerService {
...
    * @return The TrackedEntityProgramOwner object
    */
-  TrackedEntityProgramOwner getTrackedEntityProgramOwner(long teiId, long programId);
+  TrackedEntityProgramOwner getTrackedEntityProgramOwner(long teId, long programId);
 
   /**
    * Get the program owner details for a tracked entity instance.
    *
-   * @param teiUid The tracked entity instance Uid
+   * @param teUid The tracked entity instance Uid
```
Comment (enricocolasante) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/trackedentity/TrackedEntityProgramOwnerService.java:91]:
```suggestion
   * @param teUid The tracked entity Uid
```

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/trackedentity/TrackedEntityProgramOwnerService.java [hunk 4]
```diff
@@ -40,79 +40,79 @@ public interface TrackedEntityProgramOwnerService {
...
    * @param orgUnitId The organisation Unit Id
    */
-  void updateTrackedEntityProgramOwner(long teiId, long programId, long orgUnitId);
+  void updateTrackedEntityProgramOwner(long teId, long programId, long orgUnitId);
 
   /**
    * Get the program owner details for a tracked entity instance.
    *
-   * @param teiId The tracked entity instance Id
+   * @param teId The tracked entity instance Id
```
Comment (enricocolasante) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/trackedentity/TrackedEntityProgramOwnerService.java:82]:
```suggestion
   * @param teId The tracked entity Id
```

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/trackedentity/TrackedEntityProgramOwnerService.java [hunk 5]
```diff
@@ -40,79 +40,79 @@ public interface TrackedEntityProgramOwnerService {
...
    */
-  void createTrackedEntityProgramOwner(long teiId, long programId, long orgUnitId);
+  void createTrackedEntityProgramOwner(long teId, long programId, long orgUnitId);
 
   /**
    * Update the owner ou for a tracked entity instance for the given program. If no owner previously
    * exist, then this method will fail.
    *
-   * @param teiId The tracked entity instance Id
+   * @param teId The tracked entity instance Id
```
Comment (enricocolasante) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/trackedentity/TrackedEntityProgramOwnerService.java:73]:
```suggestion
   * @param teId The tracked entity Id
```

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/trackedentity/TrackedEntityProgramOwnerService.java [hunk 6]
```diff
@@ -40,79 +40,79 @@ public interface TrackedEntityProgramOwnerService {
...
    */
-  void updateTrackedEntityProgramOwner(String teiUid, String programUid, String orgUnitUid);
+  void updateTrackedEntityProgramOwner(String teUid, String programUid, String orgUnitUid);
 
   /**
    * Assign an orgUnit as the owner for a tracked entity instance for the given program. If another
    * owner already exist then this method would fail.
    *
-   * @param teiId The Id of the tracked entity instance
+   * @param teId The Id of the tracked entity instance
```
Comment (enricocolasante) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/trackedentity/TrackedEntityProgramOwnerService.java:63]:
```suggestion
   * @param teId The Id of the tracked entity
```

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/trackedentity/TrackedEntityProgramOwnerService.java (hunk 1 above)
Comment (enricocolasante) [reply dhis-2/dhis-api/src/main/java/org/hisp/dhis/trackedentity/TrackedEntityProgramOwnerService.java:43]: There are also some lost `instance` in the description of the JavaDocs

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/trackedentity/TrackedEntityProgramOwnerStore.java [hunk 7]
```diff
@@ -38,30 +38,30 @@ public interface TrackedEntityProgramOwnerStore extends GenericStore<TrackedEnti
   String ID = TrackedEntityProgramOwnerStore.class.getName();
 
   /**
-   * Get tracked entity program owner entity for the tei-program combination.
+   * Get tracked entity program owner entity for the te-program combination.
    *
-   * @param teiId The tracked entity instance id.
+   * @param teId The tracked entity instance id.
```
Comment (enricocolasante) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/trackedentity/TrackedEntityProgramOwnerStore.java:43]:
```suggestion
   * @param teId The tracked entity id.
```

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/trackedentity/TrackedEntityProgramOwnerStore.java [hunk 8]
```diff
@@ -38,30 +38,30 @@ public interface TrackedEntityProgramOwnerStore extends GenericStore<TrackedEnti
...
    */
-  TrackedEntityProgramOwner getTrackedEntityProgramOwner(long teiId, long programId);
+  TrackedEntityProgramOwner getTrackedEntityProgramOwner(long teId, long programId);
 
   /**
-   * Get all Tracked entity program owner entities for the list of teis.
+   * Get all Tracked entity program owner entities for the list of tracked entities.
    *
-   * @param teiIds The list of tracked entity instance ids.
+   * @param teIds The list of tracked entity instance ids.
```
Comment (enricocolasante) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/trackedentity/TrackedEntityProgramOwnerStore.java:52]:
```suggestion
   * @param teIds The list of tracked entity ids.
```

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/trackedentity/TrackedEntityProgramOwnerStore.java [hunk 9]
```diff
@@ -38,30 +38,30 @@ public interface TrackedEntityProgramOwnerStore extends GenericStore<TrackedEnti
...
    */
-  List<TrackedEntityProgramOwner> getTrackedEntityProgramOwners(List<Long> teiIds);
+  List<TrackedEntityProgramOwner> getTrackedEntityProgramOwners(List<Long> teIds);
 
   /**
-   * Get all Tracked entity program owner entities for the list of teis and program.
+   * Get all Tracked entity program owner entities for the list of te and program.
    *
-   * @param teiIds The list of tracked entity instance ids.
+   * @param teIds The list of tracked entity instance ids.
```
Comment (enricocolasante) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/trackedentity/TrackedEntityProgramOwnerStore.java:60]:
```suggestion
   * @param teIds The list of tracked entity ids.
```

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/trackedentity/TrackedEntityQueryParams.java [hunk 10]
```diff
@@ -867,7 +867,7 @@ public TrackedEntityQueryParams setSkipPaging(boolean skipPaging) {
     return this;
   }
 
-  public int getMaxTeiLimit() {
+  public int getMaxTeLimit() {
     return maxTeiLimit;
```
Comment (enricocolasante) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/trackedentity/TrackedEntityQueryParams.java:871]: Can we rename the field itself as well?

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/trackedentity/TrackedEntityService.java [hunk 11]
```diff
@@ -231,25 +231,25 @@ void validateSearchScope(TrackedEntityQueryParams params, boolean isGridSearch)
...
+   * @param uid Event UID to check for
    * @return true/false depending on result
    */
   boolean trackedEntityExistsIncludingDeleted(String uid);
 
   /**
    * Returns UIDs of existing tracked entities (including deleted) from the provided UIDs
    *
-   * @param uids TEI UIDs to check
+   * @param uids TE UIDs to check
```
Comment (enricocolasante) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/trackedentity/TrackedEntityService.java:252]: There is a TEIs in the return annotation

File: dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentity/hibernate/HibernateTrackedEntityStore.java [hunk 12]
```diff
@@ -211,7 +210,7 @@ public List<Long> getTrackedEntityIds(TrackedEntityQueryParams params) {
     List<Long> ids = new ArrayList<>();
 
     while (rowSet.next()) {
-      ids.add(rowSet.getLong("teiid"));
+      ids.add(rowSet.getLong("ted"));
```
Comment (enricocolasante) [inline dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentity/hibernate/HibernateTrackedEntityStore.java:213]:
```suggestion
      ids.add(rowSet.getLong("teid"));
```

Comment (enricocolasante) [reply dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentity/hibernate/HibernateTrackedEntityStore.java:213]: And all the other places in the class

File: dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentity/hibernate/HibernateTrackedEntityStore.java [hunk 13]
```diff
@@ -925,7 +924,7 @@ private String getFromSubQueryEnrollmentConditions(
     }
 
     program
-        .append("WHERE PI.trackedentityid = TEI.trackedentityid ")
+        .append("WHERE PI.trackedentityid = TE.trackedentityid ")
```
Comment (enricocolasante) [inline dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentity/hibernate/HibernateTrackedEntityStore.java:927]:
```suggestion
        .append("WHERE EN.trackedentityid = TE.trackedentityid ")
```

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/EventQuery.java [hunk 14]
```diff
@@ -95,32 +88,4 @@ public String getColumnName() {
           "getColumnName can only be invoked on TableColumn or Function");
     }
   }
-
```
Comment (enricocolasante) [inline dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/EventQuery.java:98]: It is not easy to understand from the PR. Where did you move this code?

File: dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/export/trackedentity/TrackedEntityFieldsParamMapper.java [hunk 15]
```diff
@@ -99,7 +99,7 @@ public TrackedEntityParams map(List<FieldPath> fields) {
         new TrackedEntityEnrollmentParams(
             fieldFilterService.filterIncludes(TrackedEntity.class, fields, FIELD_ENROLLMENTS),
             enrollmentParams);
-    params = params.withTeiEnrollmentParams(teiEnrollmentParams);
+    params = params.withTeEnrollmentParams(teiEnrollmentParams);
```
Comment (enricocolasante) [inline dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/export/trackedentity/TrackedEntityFieldsParamMapper.java:102]:
```suggestion
    params = params.withTeEnrollmentParams(teEnrollmentParams);
```

**Review by enricocolasante — COMMENTED** [review]
Thanks!
The task of renaming something that was around for so long is not easy and (almost) never ending.
I just found some forgotten instance here in there.
And one piece of code that looking at the PR seem that disappeared.

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/trackedentity/TrackedEntityProgramOwnerService.java (hunk 1 above)
Comment (lucaCambi77) [reply dhis-2/dhis-api/src/main/java/org/hisp/dhis/trackedentity/TrackedEntityProgramOwnerService.java:43]: yes good spot

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/EventQuery.java (hunk 14 above)
Comment (lucaCambi77) [reply dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/EventQuery.java:98]: It is unused so I've removed it. We can still restore it if we need it. I guess it has been moved in the past from /event/EventQuery to trackedEntity/aggregates/query/EventQuery

**Review by enricocolasante — APPROVED** [review]

**Pull request merged**